Update docs for OSX use cases (#917)
zmkfirmware/zmk: docs/docs/development/new-shield.md

of this guide, replace the `app/` directory in the ZMK main repository with the `config/` directory in your
user config repository. For example, `app/boards/shields/<keyboard_name>` should now be
`config/boards/shields/<keyboard_name>`.

> If you are compiling locally, you will still compile from within `zmk/app`, and you will need to provide the `DZMK_CONFIG` parameter to `west`, e.g., `-DZMK_CONFIG=/path/to/zmk-config/config`, where `/path/to` is the absolute path to your `zmk_config` directory.

Reviewer: This was helpful for me. I would suggest perhaps specifying that the parameter should go after the `--` in the given west build command.

The updated new default values should always be wrapped inside a conditional on the shield config name defined in the `Kconfig.shield` file. Here's the simplest example file.

:::warning

> Do not make the keyboard name too long, otherwise the bluetooth advertising might fail and you will not be able to find your keyboard from your laptop / tablet. The recommended length is 16 characters or less.

Reviewer 2: ```suggestion
The keyboard name must be 16 characters or less. With longer names the bluetooth advertising might fail and you will not be able to find your keyboard from your laptop / tablet.
```

I found with a longer name the code won't compile on an assert. This is not a recommendation but a requirement.
